### PR TITLE
cmd/snap: fix test suite (no Exit(0) on tests!)

### PR DESCRIPTION
**Pull request by niemeyer**

### CODE REVIEW

File: cmd/snap/cmd_sign_build_test.go [hunk 1]
```diff
@@ -65,7 +65,7 @@ func (s *SnapSignBuildSuite) TestSignBuildMissingKey(c *C) {
 
 	_, err := snap.Parser().ParseArgs([]string{"sign-build", snapFilename, "--developer-id", "dev-id1", "--snap-id", "snap-id-1"})
 	c.Assert(err, NotNil)
-	c.Check(err.Error(), Equals, "cannot use default key: cannot find key named \"default\" in GPG keyring")
+	c.Check(err.Error(), Equals, "cannot use \"default\" key: cannot find key named \"default\" in GPG keyring")
```
Comment (zyga) [inline cmd/snap/cmd_sign_build_test.go]: Thanks for fixing this!

File: cmd/snap/main_test.go [hunk 2]
```diff
@@ -142,33 +149,26 @@ func (s *SnapSuite) TestVersionOnClassic(c *C) {
...
-	c.Assert(err, IsNil)
-	c.Assert(s.Stdout(), Equals,
-		`snap    7.89
-snapd   7.89
-series  56
-ubuntu  12.34
-`)
+	restore := mockArgs("snap", "--version")
+	defer restore()
+	restore = mockVersion("4.56")
```
Comment (zyga) [inline cmd/snap/main_test.go]: While I see that this works correctly I'm wondering why you wanted to change this. We already mock the server interaction that gives us the version string, don't we?

**Comment by zyga** [comment]
Looks good, I've got one question but it doesn't change the overall +1 feeling.

I wonder if there's anything out of this that we'd like to contribute back to `go-flags`, it feels less-than-great to have to jump through hoops to perform this testing.

File: cmd/snap/main_test.go (hunk 2 above)
Comment (niemeyer) [inline cmd/snap/main_test.go]: If the tests were running at all you'd notice that this string actually _comes from the client._ :)

That makes sense, right?  We don't ask the _daemon_ what the _client_ version is.

Comment (niemeyer) [inline cmd/snap/main_test.go]: Note that the Exit(0) was happening inside RunMain on these functions, so the checks were never touched at all before.

**Comment by zyga** [comment]
This does `panic(nil)` on some tests: https://travis-ci.org/snapcore/snapd/jobs/157702777#L516

File: cmd/snap/main_test.go (hunk 2 above)
Comment (zyga) [inline cmd/snap/main_test.go]: Ah! I see. Man I feel so stupid now, having written those tests in the first place

**Pull request merged**